ext/msgpack/buffer.h

```diff
@@ -456,7 +460,11 @@ static inline VALUE msgpack_buffer_read_top_as_string(msgpack_buffer_t* b, size_
 
 #ifdef HAVE_RB_ENC_INTERNED_STR
     if (will_be_frozen) {
-        result = rb_enc_interned_str(b->read_buffer, length, utf8 ? rb_utf8_encoding() : rb_ascii8bit_encoding());
+        if (RB_ENC_INTERNED_STR_NULL_CHECK && length == 0) {
+            result = rb_enc_interned_str("", length, utf8 ? rb_utf8_encoding() : rb_ascii8bit_encoding());
```

Comment:
If we have just two frozen, interned instances of empty strings in these two encodings should they be store it somewhere and return it rather than running this code each time?

Reply:
Yeah, would be worth memoizing it in the `init`.

Reply:
Actually I remember now. This function is inline in a header, I don't know how to have a global variable here. Using `static` wouldn't work.